### ISSUE TITLE
Unable to install influxdb from source code on ubuntu 16

### ISSUE THREAD

**Issue by ManojKumarChauhan** [influxdata/influxdb#22986]
I tried to install influxdb on ununtu 16 with  go version 16 and 17 both but it's always failed.

I tried it using following 2 ways
1. go get github.com/influxdata/influxdb
2. mkdir github.com/influxdata
git clone https://github.com/influxdata/influxdb.git

Tried to compile using make, go install ./../ and go generate ./... but always get either of the following errors:

1. 
# pkg-config --cflags  -- flux
Package flux was not found in the pkg-config search path.
Perhaps you should add the directory containing `flux.pc'
to the PKG_CONFIG_PATH environment variable
No package 'flux' found
pkg-config: exit status 1

2.
./pkg-config.sh --cflags  -- flux
./pkg-config.sh: fork/exec ./pkg-config.sh: no such file or directory
Makefile:4: recipe for target 'build' failed
make: *** [build] Error 2

I set flux dir path in PKG_CONFIG_PATH but it didn't solve the problem.

**Comment by wbaker85** [comment]
Once you have installed the pre-reqs from https://github.com/influxdata/influxdb/blob/master/CONTRIBUTING.md#building-from-source, you should able to simply run `make` from the repository root. `make` will handle everything related to `pkg-config` if you have installed all of the pre-reqs. I build like this several times a day and just confirmed that it works on a fresh ubuntu installation.

**Comment by ManojKumarChauhan** [comment]
yes some dependency was missing. Thanks it's working as per the given link.

How to compile it for go version 1.16 or 1.15 without using gvm.

**Comment by wbaker85** [comment]
You'll need to use the version of go listed in `go.mod` for it to build: https://github.com/influxdata/influxdb/blob/f78c189211de559dc8132131969d495add118f97/go.mod#L3

As new versions of go are released, we frequently make use of the new features, so you won't be able to build with previous versions.

**Comment by ManojKumarChauhan** [comment]
thanks
